assets/js/Common/ServerTime.js

```diff
@@ -0,0 +1,40 @@
+const globalConfig = document.querySelector('meta[name="server-time-config"]');
+const url = globalConfig?.getAttribute('data-url') || '/api/server-time';
+const defaultMaxJitter = parseInt(globalConfig?.getAttribute('data-max-jitter') || 1000);
+const defaultDelay = parseInt(globalConfig?.getAttribute('data-delay') || 100);
+let offsetMs = 0;
+
+sync();
+
+/**
+ * @param {Number} retries
+ * @param {Number} delay
+ * @param {Number} maxJitter
+ */
+export async function sync(retries = 5, delay = defaultDelay, maxJitter = defaultMaxJitter) {
+    try {
+        const requestTime = Date.now();
+        const response = await fetch(url);
+        if (!response.ok) throw new Error();
+        const serverTime = await response.json();
+        const roundTripTime = Date.now() - requestTime;
+        const timeWhenServerCreatedServerTime = requestTime + roundTripTime / 2;
+        offsetMs = serverTime - timeWhenServerCreatedServerTime;
```

Comment:
If the offset is below a certain threshold, e.g. 100ms, maybe it can be set to 0.